pkg/util/metricsutil/common.go

```diff
@@ -67,22 +66,26 @@ func RegisterMetrics() error {
 	}
 
 	timeoutSec := time.Duration(cfg.PDClient.PDServerTimeout) * time.Second
+	// Note: for NextGen, pdCli is created to init the metrics' const labels
 	pdCli, err := pd.NewClient(componentName, pdAddrs, pd.SecurityOption{
 		CAPath:   cfg.Security.ClusterSSLCA,
 		CertPath: cfg.Security.ClusterSSLCert,
 		KeyPath:  cfg.Security.ClusterSSLKey,
-	}, opt.WithCustomTimeoutOption(timeoutSec))
+	}, opt.WithCustomTimeoutOption(timeoutSec), opt.WithMetricsLabels(metrics.GetConstLabels()))
 	if err != nil {
 		return err
 	}
 	defer pdCli.Close()
 
-	keyspaceMeta, err := getKeyspaceMeta(pdCli, cfg.KeyspaceName)
-	if err != nil {
-		return err
+	if kerneltype.IsNextGen() {
+		registerMetrics(nil) // metrics' const label already set
+	} else {
+		keyspaceMeta, err := getKeyspaceMeta(pdCli, cfg.KeyspaceName)
+		if err != nil {
+			return err
+		}
+		registerMetrics(keyspaceMeta)
```

Comment:
Could we move the call to `metrics.SetConstLabels("keyspace_id"...)` earlier in the logic(line59-61)? This way, we can handle the next-gen keyspace info setup in the same place. However, I'm not sure if it's feasible due to the different initMetrics calls.

Reply:
Actually, for next gen, we should set const labels first, then we can use const labels to init pdCli; while for non-next gen, we should get pdCli first, then get keyspaceMeta using pdCli, and finally set const labels...

Reply:
Well. 
Then, should we consider extracting these lines of code into a function, so that the related code only needs to be modified in one place